tests/manage/pv_services/test_daemon_kill_during_pvc_pod_creation_and_io.py

```diff
@@ -0,0 +1,392 @@
+                resource=pod_obj, state=constants.STATUS_RUNNING
+            )
+            pod_obj.reload()
+        log.info("Verified: All new pods are Running.")
+
+        # Verify IO
+        log.info("Fetching IO results from IO pods.")
+        for pod_obj in io_pods:
+            fio_result = pod_obj.get_fio_results()
+            err_count = fio_result.get('jobs')[0].get('error')
+            assert err_count == 0, (
+                f"FIO error on pod {pod_obj.name}. FIO result: {fio_result}"
+            )
+            log.info(f"IOPs after FIO on pod {pod_obj.name}:")
+            log.info(
+                f"Read: {fio_result.get('jobs')[0].get('read').get('iops')}"
+            )
+            log.info(
+                f"Write: {fio_result.get('jobs')[0].get('write').get('iops')}"
+            )
+        log.info("Verified IO result on IO pods.")
+
+        all_pod_objs = io_pods + pod_objs_new
+
+        # Fetch volume details from pods for the purpose of verification
+        node_pv_dict = {}
+        for pod in all_pod_objs:
+            pod_info = pod.get()
+            node = pod_info['spec']['nodeName']
+            pvc = pod_info['spec']['volumes'][0]['persistentVolumeClaim']['claimName']
+            for pvc_obj in pvc_objs:
+                if pvc_obj.name == pvc:
+                    pvc_obj.reload()
+                    pv = pvc_obj.backed_pv
+                    break
+            if node in node_pv_dict:
+                node_pv_dict[node].append(pv)
+            else:
+                node_pv_dict[node] = [pv]
+
+        # Delete pods
+        for pod_obj in all_pod_objs:
+            pod_obj.delete(wait=False)
+
+        # Verify pods are deleted
+        for pod_obj in all_pod_objs:
+            pod_obj.ocp.wait_for_delete(resource_name=pod_obj.name)
+
+        # Verify number of 'resource_name' type pods
+        final_resource_pods_num = len(pod_functions[resource_name]())
+        assert final_resource_pods_num == resource_pods_num, (
+            f"Total number of {resource_name} pods is not matching with "
+            f"initial value. Total number of pods before daemon kill: "
+            f"{resource_pods_num}. Total number of pods present now: "
+            f"{final_resource_pods_num}"
+        )
+
+        # Verify volumes are unmapped from nodes after deleting the pods
+        node_pv_mounted = helpers.verify_pv_mounted_on_node(node_pv_dict)
+        for node, pvs in node_pv_mounted.items():
+            assert not pvs, (
+                f"PVs {pvs} is still present on node {node} after "
+                f"deleting the pods."
+            )
+        log.info(
+            "Verified: mount points are removed from nodes after deleting "
+            "the pods"
+        )
+
+        # Set volume mode on PVC objects
+        for pvc_obj in pvc_objs_new:
+            pvc_info = pvc_obj.get()
+            setattr(pvc_obj, 'volume_mode', pvc_info['spec']['volumeMode'])
```

Comment:
when you do a setattr(), does it do a reload of pvc_obj() here?

Reply:
It is not doing reload. Other attribute values remain the same.
No need to do reload here. volume_mode is used only in line 156

Reply:
so how does it get used?

Reply:
It will be an additional attribute of pod_obj. The check in line 156 is done after setting volume_mode attribute.